### PR TITLE
change makeView, create automatically searchViewArch when an arch exists

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/tests/views/helpers.js [hunk 1]
```diff
@@ -23,11 +23,13 @@ export async function makeView(params) {
...
-    }
-    if (props.searchViewArch && !props.searchViewFields) {
-        props.searchViewFields = Object.assign({}, props.fields || defaultFields);
+    if (props.arch) {
+        if (!props.fields) {
+            props.fields = Object.assign({}, defaultFields);
+        }
+        props.searchViewArch = props.searchViewArch || "<search/>";
+        props.searchViewFields =
+            props.searchViewFields || Object.assign({}, props.fields || defaultFields);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/helpers.js]: props.searchViewFields || Object.assign({}, props.fields);

**Review by aab-odoo — COMMENTED** [review]
could we remove as well the "views" key set in serverData in beforeEach?

File: addons/web/static/tests/views/graph_view_tests.js [hunk 2]
```diff
@@ -1846,7 +1846,11 @@ QUnit.module("Views", (hooks) => {
                     <field name="bar"/>
                 </graph>
             `,
-            views: [[false, "search"]],
+            searchViewArch: `
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/graph_view_tests.js:1849]: is it possible to remove the search view arch in serverData?

Comment (jpp-odoo) [reply addons/web/static/tests/views/graph_view_tests.js:1849]: It can be removed, but for the moment it's used on other tests (it's not a simple remove).

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**